docs/meshstack.configuration.md

```diff
@@ -134,11 +135,203 @@ When a user is invited to a customer there are several configurations in order t
+      Attention: This check is only performed on the first attempt when a user is added/invited
+      to a customer. If this check is configured after some users were initially added to a
+      customer they are not detected as guest users anymore.
+
+        guest-property:
+            The AAD's custom attribute which is checked against the guest-value.
+
+        guest-value:
+            If the AAD custom attribute matches this value the user is considered to be a guest.
+    -}
+      { guest-property : Text, guest-value : Text }
+```
+<!--Example-->
+```dhall
+let example
+    : AzureGuestDetection
+    = { guest-property = "guest-property", guest-value = "is-guest" }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+<!--snippet:mesh.meshfed.identity.azure.euid#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let AzureEuidExtensionSchema =
+    {-
+      When adding/inviting a new user to a customer a custom attribute property from the users AAD
+      extension schema can be used to fill in his euid.
+
+      Attention: This check is only performed on the first attempt when a user is added/invited
+      to a customer. If this check is configured after some users were initially added to a
+      customer their euid's are not udpated anymore.
+
+        euidExtensionSchemaProperty:
+            The name of the extension schema property which is used as EUID value.
+    -}
+      { euidExtensionSchemaProperty : Text
+      , euidExtensionSchemaIdentifier : Text
+      }
+```
+<!--Example-->
+```dhall
+let example
+    : AzureEuidExtensionSchema
+    = { euidExtensionSchemaProperty = "euid-schema-property"
+      , euidExtensionSchemaIdentifier = "euid-schema-identfifier"
+      }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+If you want to use Google Cloud Directory in order to perform additional user setup steps you must configure the principal like so:
+
+<!--snippet:mesh.meshfed.identity.gcd.creds#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let GcdCreds =
+    {-
+      Setting this configuration enables the use of an GCD as a user lookup source to allow
+      autocomplete of user information when adding new users to customers.
+
+        domain:
+            The domain used for cloud identity directory-groups created and managed by meshStack.
+            meshStack maintains separate groups for each meshProject role on each managed GCP project.
+
+        customer-id:
+            The client id of the service principal
+
+        service-account-credentials-b64:
+            The credentials of the service principal
+    -}
+      { domain : Text
+      , customer-id : Text
+      , service-account-credentials-b64 : Secret.Type
+      }
+```
+<!--Example-->
+```dhall
+let example
+    : GcdCreds
+    = { domain = "example.com"
+      , customer-id = "customer-id"
+      , service-account-credentials-b64 =
+          Secret.fromAnsible "gcp_credentials"
+      }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
```

Comment:
f: As with the Azure part, this duplicates https://docs.meshcloud.io/docs/meshstack.configuration.html#configure-for-google-cloud-directory.

Reply:
unified the GCP stuff as well.